forward/forwarder.go

```diff
@@ -50,6 +50,7 @@ type Options struct {
 	RerouteRetries bool
 	RetrySchedule  []time.Duration
 	Timeout        time.Duration
+	Headers        []byte
```

Comment:
*I think* this is a generic approach that works when forwarding raw requests, where the headers can be anything.

Maybe @prashantv can shed some light on why he chose to use `[]byte` in his original PR.